### PR TITLE
Gh12 ut control building issues from testing suite

### PR DESCRIPTION
**Pull request by kanjoe24**

### CODE REVIEW

File: include/ut_kvp.h [hunk 1]
```diff
@@ -25,6 +25,8 @@
 #include <stdint.h>
 #include <stdio.h>
 
+#include <ut_log.h>
```
Comment (Ulrond) [inline include/ut_kvp.h:28]: remove this.

File: src/ut_control_plane.c [hunk 2]
```diff
@@ -29,8 +29,8 @@
 /* External libraries */
 #include <libwebsockets.h>
 
-#define UT_CONTROL_PLANE_ERROR(f_, ...) printf((f_), ##__VA_ARGS__)
-#define UT_CONTROL_PLANE_DEBUG(f_, ...) (void)0
+#define UT_CONTROL_PLANE_ERROR(f_, ...) UT_LOG_ERROR((f_), ##__VA_ARGS__)
```
Comment (Ulrond) [inline src/ut_control_plane.c:32]: include <ut_log.h> - here.

File: src/ut_kvp.c [hunk 3]
```diff
@@ -22,16 +22,10 @@
 #include <string.h>
 #include <limits.h>
 #include <unistd.h>
+#include <assert.h>
 
 /* Application Includes */
 #include <ut_kvp.h>
```
Comment (Ulrond) [inline src/ut_kvp.c:28]: `#include <ut_log.h>` - required

**Review by Ulrond — CHANGES_REQUESTED** [review]
minor changes.

File: tests/Makefile [hunk 4]
```diff
@@ -30,6 +30,7 @@
 ROOT_DIR:=$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
 BIN_DIR := $(ROOT_DIR)/bin
 TOP_DIR := $(ROOT_DIR)
+LIBWEBSOCKET_LIB_DIR = $(ROOT_DIR)/../framework/libwebsockets-4.3.3/build/lib
```
Comment (kanjoe24) [inline tests/Makefile:33]: Added this as hack as of now, will remove later

File: include/ut_kvp.h
Comment (Ulrond) [inline include/ut_kvp.h:1]: Add void* into the callback

```c
typedef void (*ut_control_callback_t)( char *key, ut_kvp_instance_t *instance, void *userData );
...
ut_control_plane_status_t UT_ControlPlane_RegisterCallbackOnMessage(
                                                                    ut_controlPlane_instance_t *pInstance,
                                                                    char *key,
                                                                    ut_control_callback_t callbackFunction
                                                                    void *userData );
```

Comment (Ulrond) [inline include/ut_kvp.h:1]: remove `void ut_kvp_print( ut_kvp_instance_t *pInstance );` and replace with `uint8_t *ut_kvp_getData( ut_kvp_instance_t *pInstance, uint32_t *pReturnedLength );`

- Upgrading the tests as required still to print but the print will be external and you will call getData instead.

Ensure that you state that the `@note: caller is responsible for freeing the memory` in the doxygen comments.

**Review by Ulrond — APPROVED** [review]
looks good

**Pull request merged**